src/app/utils/field-validation/validators/numberValidator.ts

```diff
@@ -59,29 +62,64 @@ const maxLengthValidator: NumberValidatorConstructor =
 const exactLengthValidator: NumberValidatorConstructor =
   (numberField) => (response) => {
     const { answer } = response
-    const { customVal } = numberField.ValidationOptions
+    const { customVal } = numberField.ValidationOptions.LengthValidationOptions
     return !customVal || answer.length === customVal
       ? right(response)
       : left(`NumberValidator:\t answer does not match custom exact length`)
   }
 
 /**
- * Returns the appropriate validation function
- * based on the number validation option selected.
+ * Returns the appropriate number length validation function
+ * based on the number length validation option selected.
  */
 const getNumberLengthValidator: NumberValidatorConstructor = (numberField) => {
-  switch (numberField.ValidationOptions.selectedValidation) {
-    case NumberSelectedValidation.Min:
+  switch (
+    numberField.ValidationOptions.LengthValidationOptions
+      .selectedLengthValidation
+  ) {
+    case NumberSelectedLengthValidation.Min:
       return minLengthValidator(numberField)
-    case NumberSelectedValidation.Max:
+    case NumberSelectedLengthValidation.Max:
       return maxLengthValidator(numberField)
-    case NumberSelectedValidation.Exact:
+    case NumberSelectedLengthValidation.Exact:
       return exactLengthValidator(numberField)
     default:
       return right
   }
 }
 
+/**
+ * Returns a validation function to check if number is
+ * within the number range specified.
+ */
+const getNumberRangeValidator: NumberValidatorConstructor =
+  (numberField) => (response) => {
+    const val = Number(response.answer)
+    const { customMin, customMax } =
+      numberField.ValidationOptions.RangeValidationOptions
+    const isWithinMinimum = !!customMin && customMin <= val
+    const isWithinMaximum = !!customMax && val <= customMax
+
+    return isWithinMinimum && isWithinMaximum
+      ? right(response)
+      : left(`NumberValidator:\t answer does not fall within specified range`)
+  }
+
+/**
+ * Returns the appropriate number validation function
+ * based on the number validation option selected.
+ */
+const getNumberValidator: NumberValidatorConstructor = (numberField) => {
+  switch (numberField.ValidationOptions.selectedValidation) {
+    case NumberSelectedValidation.Length:
+      return getNumberLengthValidator(numberField)
+    case NumberSelectedValidation.Range:
+      return getNumberRangeValidator(numberField)
```

Comment:
Honestly, I would not be opposed to saving ourselves the verbosity of adding an extra layer of calling the `getNumberLengthValidator` and just embed that switch into this switch, then get rid of `getNumberLengthValidator` completely:

```suggestion
    case NumberSelectedValidation.Length:
      switch (numberField.ValidationOptions.LengthValidationOptions.selectedLengthValidation) {
        case NumberSelectedLengthValidation.Min:
            return minLengthValidator(numberField)
        case NumberSelectedLengthValidation.Max:
            return maxLengthValidator(numberField)
        case NumberSelectedLengthValidation.Exact:
            return exactLengthValidator(numberField)
        default:
             return right
      }
    case NumberSelectedValidation.Range:
      return rangeValidator(numberField)
```

Up to you though!

Reply:
In my opinion, the extra verbosity in this case makes the function a lot easier to understand, so I would prefer if we kept it!